### PR TITLE
Replace docker-build script with multi-stage build

### PR DESCRIPTION
**Pull request by furuholm**
Uses docker multi stage builds [1] instead docker-build script. The resulting image should be identical to running scripts/docker-build and docker build separately.

[1] https://docs.docker.com/engine/userguide/eng-image/multistage-build/

### CODE REVIEW

**Comment by furuholm** [comment]
Fixes #1074

File: .dockerignore [hunk 1]
```diff
@@ -1,3 +1,14 @@
 *
```
Comment (ericchiang) [inline .dockerignore]: We actually don't need to ignore anything now, right?

Comment (furuholm) [reply .dockerignore]: Not really.

**Comment by furuholm** [comment]
No changes to the makefile are required (AFAIK)

File: .dockerignore (hunk 1 above)
Comment (furuholm) [reply .dockerignore]: I just tried to follow the convention of the current dockerfile. Figured you maybe had reasons for wanting to ignore everything but the necessary files.

**Comment by ericchiang** [comment]
This PR removes a script referenced in the Makefile https://github.com/coreos/dex/blob/64c97c73080bfc951d5f01906257a633bd164909/Makefile#L61-L63

I'd expect changes to that Makefile as part of the PR.

File: .dockerignore (hunk 1 above)
Comment (ericchiang) [reply .dockerignore]: Can we just remove the `.dockerignore` file then? It'd be weird to have to always update this file every time we add a new top level directory.

Comment (furuholm) [reply .dockerignore]: Will do

**Comment by furuholm** [comment]
My mistake! I'll go through the makefile and make sure it is up to date.

**Comment by furuholm** [comment]
Makefile updated. Did not remove the docker ignore file though. Left the bin directory in there as it gets quite large.

**Comment by ericchiang** [comment]
This worked for me locally. Thanks for the change! Can you please squash your commits then we can merge?

**Comment by furuholm** [comment]
Great! Rebased and squashed.

**Pull request merged**